src/WebJobs.Extensions.Http/ClaimsIdentityHelper.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Json;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    /// <summary>
+    /// Provides methods to grab the ClaimsPrincipal from an HTTP Request trigger
+    /// </summary>
+    public static class ClaimsIdentityHelper
+    {
+        // Shared serializer instance. This is safe for multi-threaded use.
+        private static readonly DataContractJsonSerializer Serializer = GetCustomSerializer();
+
+        /// <summary>
+        /// Retrieves a serialized ClaimsIdentity object from the http request of an HTTP Trigger
+        /// </summary>
+        /// <param name="request">The request message from the HTTP Trigger</param>
+        /// <param name="identityHeaderName">The name of the header the identity is stored on.</param>
+        /// <returns></returns>
+        public static ClaimsIdentity GetIdentityFromHttpRequest(HttpRequest request, string identityHeaderName)
+        {
+            if (request == null)
+            {
+                return null;
```

Comment:
>return null; [](start = 16, length = 12)

Would it make more sense to `throw ArgumentException(nameof(request))` instead of returning `null`?